src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentLoadContext.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+using System.Runtime.Loader;
+
+namespace Internal.Runtime.InteropServices
+{
+    internal class ComponentLoadContext : AssemblyLoadContext
```

Comment:
I think a better name would be `IsolatedComponentLoadContext` to make it clear that this ALC implementation provides full isolation.